sdk/keyvault/Azure.Security.KeyVault.Keys/src/JsonWebKey.cs

```diff
@@ -305,7 +305,7 @@ public RSA ToRSA(bool includePrivateParameters = false)
                 Modulus = TrimBuffer(N),
             };
 
-            if (includePrivateParameters)
+            if (includePrivateParameters && HasPrivateKey)
```

Comment:
Should add some unit tests for this change.

Reply:
Added issue https://github.com/Azure/azure-sdk-for-net/issues/7844 to track this.